### DISCUSSION TITLE
Welcome to neuron Discussions!

### DISCUSSION THREAD

**Discussion by CrazyWisdom** [emqx/neuron#230] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by wlq965** [comment]
你好，我按照开发流程编译neuron工程的时候，报错，
cd neuron && mkdir build && cd build
cmake .. && make
CMake Error: File /home/wlq/user/Neuron/neuron/extern/uthash/include/uthash.h does not exist.
CMake Error at CMakeLists.txt:35 (configure_file):
  configure_file Problem configuring file


CMake Error: File /home/wlq/user/Neuron/neuron/extern/uthash/include/utarray.h does not exist.
CMake Error at CMakeLists.txt:36 (configure_file):
  configure_file Problem configuring file

**Reply by wlq965** [reply]
@CrazyWisdom 您帮忙看看

**Reply by wlq965** [reply]
上面的问题已解决

**Reply by wlq965** [reply]
新的问题，openssl版本问题，
/home/wlq/user/Neuron/neuron/src/connection/mqtt_c_client.c:188:5: error: ‘ERR_load_BIO_strings’ is deprecated: Since OpenSSL 3.0 [-Werror=deprecated-declarations]
  188 |     ERR_load_BIO_strings();
      |     ^~~~~~~~~~~~~~~~~~~~
@CrazyWisdom 您使用的openssl版本是哪个版本

**Reply by wlq965** [reply]
openssl 3.0的版本

**Reply by fengzeroz** [reply]
@wlq965 目前依赖openssl 1.1.1版本

**Reply by WZSHUO1207** [reply]
@wlq965 您好，第一个问题怎么解决的？

**Reply by fengzeroz** [reply]
> @wlq965 您好，第一个问题怎么解决的？

update submodule

**Comment by SCHMOTT22** [comment]
I have been doing my research on this project. I was wondering if you could give me any input to help or put me in the write direction?

**Reply by fengzeroz** [reply]
> I have been doing my research on this project. I was wondering if you could give me any input to help or put me in the write direction?

In the next 2 months, we will provide sdk, sdk related documents, and examples of developing plugins based on neuron. We will also provide some architectural design documents later.

**Reply by SCHMOTT22** [reply]
> > I have been doing my research on this project. I was wondering if you could give me any input to help or put me in the write direction?
> 
> 
> 
> In the next 2 months, we will provide sdk, sdk related documents, and examples of developing plugins based on neuron. We will also provide some architectural design documents later.

Ok. That sounds great. Where do I find these docs?

**Reply by SCHMOTT22** [reply]
> > > I have been doing my research on this project. I was wondering if you could give me any input to help or put me in the write direction?
> > 
> > 
> > 
> > In the next 2 months, we will provide sdk, sdk related documents, and examples of developing plugins based on neuron. We will also provide some architectural design documents later.
> 
> Ok. That sounds great. Where do I find these docs?

I'm sorry I have been teaching myself all of this for a couple years. Which doc do u mean?

**Comment by SCHMOTT22** [comment]
How do you get you miner or set your miner up to receive and send data packets?

**Reply by fengzeroz** [reply]
> How do you get you miner or set your miner up to receive and send data packets?

Sorry, I did not understand what you said.

**Comment by lauri33141** [comment]
Any good guidelines how to install Neuron open source edition ? What changes one have to make for that ? insatll from packet possible ?
At this point we managed to install without licence, but when adding modbus device got error 2401 licence invalid. Managed to add mqtt device though.
